content/en/network_monitoring/devices/troubleshooting.md

```diff
@@ -12,91 +12,165 @@ further_reading:
+The output should look similar to the following: 
+
+   ```
+   snmp
+   ----
+     Instance ID: snmp:default:1.2.3.4.1:9a2df638d3ba38d6 [ERROR]
+     Configuration Source: file:/etc/datadog-agent/conf.d/snmp.d/conf.yaml
+     Total Runs: 1
+     Metric Samples: Last Run: 6, Total: 6
+     Events: Last Run: 0, Total: 0
+     Network Devices Metadata: Last Run: 1, Total: 1
+     Service Checks: Last Run: 1, Total: 1
+     Average Execution Time : 0s
+     Last Execution Date : 2024-11-13 13:12:09 PST / 2024-11-13 21:12:09 UTC (1731532329000)
+     Last Successful Execution Date : Never
+     Error: <b>&lt;ERROR MESSAGE&gt;</b> 
+     No traceback
+   ```
+
+2. If your device is not listed and you are using Autodiscovery, it likely means the Agent could not connect to your device. Try running an `snmp walk` on the device's admin IP. 
+
+   **Note**: Provide your credentials directly in the CLI. If credentials aren't provided, the Agent attempts to locate them in your running Agent configuration files. 
+
+   **Linux**: 
+   For SNMP v2:
+   ```
+   sudo -u dd-agent datadog-agent snmp walk <IP Address> -C <COMMUNITY_STRING>
+   ```
+   For SNMP v3:
+    ```
+    sudo -u dd-agent datadog-agent snmp walk <IP Address> -A <AUTH_KEY> -a <AUTH_PROTOCOL> -X <PRIV_KEY> -x <PRIV_PROTOCOL>
+    ```
+    **Windows**:   
+    ```
+    agent snmp walk <IP Address>[:Port]
 
-SNMP - Simple network management protocol
-: A network protocol that is used to collect information about bare metal networking gear.
+    Example:           
+    agent.exe snmp walk  10.143.50.30 1.3.6 
+    ```
 
-OID - Object identifier
-: A unique ID or address on a device that when polled returns the response code of that value. For example, OIDs are CPU or device fan speed.
+    Refer to your vendor specific documentation for additional information on running these commands.
 
-sysOID - System object identifier
-: A specific address that defines the device type. All devices have a unique ID that defines it. For example, the Meraki base sysOID is `1.3.6.1.4.1.29671`.
+3. If either the SNMP status or Agent walk show an error, it could indicate one of the following issues:
 
-MIB - Managed information base
-: A database or list of all the possible OIDs and their definitions that are related to the MIB. For example, the `IF-MIB` (interface MIB) contains all the OIDs for descriptive information about a device's interface.
+   **Unreachable or misconfigured device**:
 
-## FAQ
+   ```
+   Error: check device reachable: failed: error reading from socket: read udp 127.0.0.1:46068->1.2.3.4:161
+   ```
+   
+   **Solution**:
+   - Log into your device and ensure that SNMP is enabled and exposed on port 161.
+   - Ensure you collector firewall allows egress:
 
-#### What SNMP versions does Datadog support?
+   **Linux only**:
```

Comment:
what parent section does this "Linux only" section belong to?  is it related to "Unreachable or misconfigured device" or is it its own issue?

in fact, I think this whole section could use more hierarchy—each solution should be a sub-section of its relevant issue

Reply:
it's related to the Unreachable section but yes.. Agree, let me rework this a bit!